c/tskit/trees.c

```diff
@@ -808,7 +808,7 @@ tsk_treeseq_get_individuals_time(const tsk_treeseq_t *self, double *output)
 
 /* Stats functions */
 
-#define GET_2D_ROW(array, row_len, row) (array + (((size_t)(row_len)) * (size_t) row))
+#define GET_2D_ROW(array, row_len, row) (array + (((size_t)(row_len)) * (size_t)(row)))
```

Comment:
This macro has bitten me a few times now. I thought I'd make it a bit safer by adding parentheses so that expressions don't have strange behavior when passed in. For instance, when passing in an offset (ie n+4), the operator precedence of the multiplication makes the final 2d row offset incorrect. I don't foresee any issues with an extra set of parentheses, but macros can be tricky.

Reply:
Ouch - good update and sorry for my macro parameter sloppiness! Feel free to change to an inline function if you like, I don't think it makes much difference in practise.